### ISSUE TITLE
[AIX] Java GUI application does not start on JA_JP locale on minimum system configuration

### ISSUE THREAD

**Issue by takiguc** [AdoptOpenJDK/openjdk-build#1239]
**Platform:**
AIX 7.1 / 7.2

hotspot
openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.4+11, mixed mode)

openj9
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.15.1, JRE 11 AIX ppc64-64-Bit Compressed References 20190717_292 (JIT enabled, AOT enabled)
OpenJ9   - 0f66c64
OMR      - ec782f2
JCL      - fa49279 based on jdk-11.0.4+11)

**Architecture:**
ppc64

**Issue**
On JA_JP (Japanese UTF-8) locale, Java GUI application does not start with following exception.
This issue happens on minimum Japanese locale configuration.
(If the system installed all TrueType fonts on AIX, user does not see this issue.)
Other CJK UTF-8 locales have same issue.

```
Exception in thread "AWT-EventQueue-0" java.lang.ClassCastException: sun.font.CompositeFont incompatible with sun.font.PhysicalFont
        at java.desktop/sun.font.SunFontManager.getDefaultPhysicalFont(SunFontManager.java:1086)
        at java.desktop/sun.font.NativeFont.createStrike(NativeFont.java:259)
        at java.desktop/sun.font.Font2D.getStrike(Font2D.java:359)
        at java.desktop/sun.font.Font2D.getStrike(Font2D.java:308)
        at java.desktop/sun.font.CompositeStrike.getStrikeForSlot(CompositeStrike.java:80)
        at java.desktop/sun.font.CompositeStrike.getFontMetrics(CompositeStrike.java:95)
        at java.desktop/sun.font.FontDesignMetrics.initMatrixAndMetrics(FontDesignMetrics.java:360)
        at java.desktop/sun.font.FontDesignMetrics.<init>(FontDesignMetrics.java:351)
        at java.desktop/sun.font.FontDesignMetrics.getMetrics(FontDesignMetrics.java:303)
        at java.desktop/sun.swing.SwingUtilities2.getFontMetrics(SwingUtilities2.java:1231)
        at java.desktop/javax.swing.JComponent.getFontMetrics(JComponent.java:1646)
        at java.desktop/javax.swing.plaf.basic.BasicGraphicsUtils.getPreferredButtonSize(BasicGraphicsUtils.java:351)
        at java.desktop/javax.swing.plaf.basic.BasicButtonUI.getPreferredSize(BasicButtonUI.java:474)
        at java.desktop/javax.swing.JComponent.getPreferredSize(JComponent.java:1680)
        at java.desktop/javax.swing.plaf.metal.MetalDesktopIconUI.installComponents(MetalDesktopIconUI.java:86)
        at java.desktop/javax.swing.plaf.basic.BasicDesktopIconUI.installUI(BasicDesktopIconUI.java:83)
        at java.desktop/javax.swing.JComponent.setUI(JComponent.java:685)
        at java.desktop/javax.swing.JInternalFrame$JDesktopIcon.setUI(JInternalFrame.java:2162)
        at java.desktop/javax.swing.JInternalFrame$JDesktopIcon.updateUI(JInternalFrame.java:2210)
        at java.desktop/javax.swing.JInternalFrame$JDesktopIcon.<init>(JInternalFrame.java:2142)
        at java.desktop/javax.swing.JInternalFrame.<init>(JInternalFrame.java:348)
        at java.desktop/javax.swing.JInternalFrame.<init>(JInternalFrame.java:280)
        at InternalFrameDemo.createInternalFramePalette(InternalFrameDemo.java:174)
        at InternalFrameDemo.<init>(InternalFrameDemo.java:120)
        at SwingSet2.preloadFirstDemo(SwingSet2.java:607)
        at SwingSet2.<init>(SwingSet2.java:183)
        at SwingSet2.lambda$main$1(SwingSet2.java:204)
        at SwingSet2$$Lambda$8.00000000123C06D0.run(Unknown Source)
        at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:704)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
        at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

To avoid this issue, need to modify lib/fontconfig.properties by following fc.sed SED script.
```
/-ucs2\.cjk_japan-0/s/-monotype-timesnewromanwt-medium-r-normal--/-monotype-wt serif j-medium-r-normal--/
/-ucs2\.cjk_korea-0/s/-monotype-timesnewromanwt-medium-r-normal--/-monotype-wt serif k-medium-r-normal--/
/-ucs2\.cjk_china-0/s/-monotype-timesnewromanwt-medium-r-normal--/-monotype-wt serif sc-medium-r-normal--/
/-ucs2\.cjk_china-0/s/-monotype-timesnewromanwt-medium-r-normal--/-monotype-wt serif sc-medium-r-normal--/
/-ucs2\.cjk_taiwan-0/s/-monotype-timesnewromanwt-medium-r-normal--/-monotype-wt serif tw-medium-r-normal--/
/-ucs2\.cjk_japan-0/s/-monotype-sansmonowt-medium-r-normal--/-monotype-wt sans duo j-medium-r-normal--/
/-ucs2\.cjk_korea-0/s/-monotype-sansmonowt-medium-r-normal--/-monotype-wt sans duo k-medium-r-normal--/
/-ucs2\.cjk_china-0/s/-monotype-sansmonowt-medium-r-normal--/-monotype-wt sans duo sc-medium-r-normal--/
/-ucs2\.cjk_taiwan-0/s/-monotype-sansmonowt-medium-r-normal--/-monotype-wt sans duo tw-medium-r-normal--/
/-ucs2\.cjk_japan-0/s/-monotype-sanswt-medium-r-normal--/-monotype-wt sans j-medium-r-normal--/
/-ucs2\.cjk_korea-0/s/-monotype-sanswt-medium-r-normal--/-monotype-wt sans k-medium-r-normal--/
/-ucs2\.cjk_china-0/s/-monotype-sanswt-medium-r-normal--/-monotype-wt sans sc-medium-r-normal--/
/-ucs2\.cjk_taiwan-0/s/-monotype-sanswt-medium-r-normal--/-monotype-wt sans tw-medium-r-normal--/
/-ucs2\.cjk_japan-0/s/\/mtsans_j\.ttf$/\/wts_j__b\.ttf/
/-ucs2\.cjk_japan-0/s/\/tnrwt_j\.ttf$/\/wt__j__b\.ttf/
/-ucs2\.cjk_japan-0/s/\/mtsansdj\.ttf$/\/wtsdj__b\.ttf/
/-ucs2\.cjk_taiwan-0/s/\/tnrwt_t\.ttf$/\/wt__tt_b\.ttf/
/-ucs2\.cjk_korea-0/s/\/mtsansdk\.ttf$/\/wtsdk__b\.ttf/
/-ucs2\.cjk_korea-0/s/\/tnrwt_k\.ttf$/\/wt__k__b\.ttf/
/-ucs2\.cjk_taiwan-0/s/\/mtsansdt\.ttf$/\/wtsdtt_b\.ttf/
/-ucs2\.cjk_china-0/s/\/mtsans_s\.ttf$/\/wts_s__b\.ttf/
/-ucs2\.cjk_china-0/s/\/tnrwt_s\.ttf$/\/wt__s__b\.ttf/
/-ucs2\.cjk_korea-0/s/\/mtsans_k\.ttf$/\/wts_k__b\.ttf/
/-ucs2\.cjk_taiwan-0/s/\/mtsans_t\.ttf$/\/wts_tt_b\.ttf/
/-ucs2\.cjk_china-0/s/\/mtsansds\.ttf$/\/wtsds__b\.ttf/
s/-ucs2\.cjk_korea-0/-iso10646-1/
s/-ucs2\.cjk_japan-0/-iso10646-1/
s/-ucs2\.cjk_taiwan-0/-iso10646-1/
s/-ucs2\.cjk_china-0/-iso10646-1/
/^filename./s/ /_/g
```
Type following commands:
```
$ cd lib
$ sed -f fc.sed fontconfig.properties.src > fontconfig.properties
```

**Comment by takiguc** [comment]
I could recreate this issue on JDK13/14.
I'll report this issue to OpenJDK.
Related issue was reported:
https://bugs.openjdk.java.net/browse/JDK-8221741

**Comment by gdams** [comment]
@takiguc can this be closed now that it has been reported upstream?

**Comment by takiguc** [comment]
Now JDK-8221741 is in final review phase.
For my tracking, could you wait for a moment until push the code?

**Comment by takiguc** [comment]
JDK-8221741 was resolved for JDK14.
The fix was ported to 11.0.6 b05 via JDK-8234797.
https://bugs.openjdk.java.net/browse/JDK-8234797
